app-start/src/main/java/com/example/android/people/ReplyReceiver.kt

```diff
@@ -25,26 +25,20 @@ import com.example.android.people.data.DefaultChatRepository
 /**
  * Handles the "Reply" action in the chat notification.
  */
-class ReplyReceiver : BroadcastReceiver() {
+
+// Extend `ReplyReceiver` from `BroadcastReceiver` class
+class ReplyReceiver {
 
     companion object {
         const val KEY_TEXT_REPLY = "reply"
     }
-
-    override fun onReceive(context: Context, intent: Intent) {
-        val repository: ChatRepository = DefaultChatRepository.getInstance(context)
-
-        val results = RemoteInput.getResultsFromIntent(intent) ?: return
-        // The message typed in the notification reply.
-        val input = results.getCharSequence(KEY_TEXT_REPLY)?.toString()
-        val uri = intent.data ?: return
-        val chatId = uri.lastPathSegment?.toLong() ?: return
-
-        if (chatId > 0 && !input.isNullOrBlank()) {
-            repository.sendMessage(chatId, input.toString(), null, null)
-            // We should update the notification so that the user can see that the reply has been
-            // sent.
-            repository.updateNotification(chatId)
-        }
-    }
```

Comment:
предлагаю код не относящийся к броадкастам оставить, но вынести в отдельный метод.
```
fun sendMessage(chatId: Long, input: String) {
    val repository: ChatRepository = DefaultChatRepository.getInstance(context)
    if (chatId > 0 && !input.isNullOrBlank()) {
            repository.sendMessage(chatId, input.toString(), null, null)
            // We should update the notification so that the user can see that the reply has been
            // sent.
            repository.updateNotification(chatId)
        }

}
```
от студента главное сделать броадкаст обработать

Reply:
да согласно эта будет удобна для студента 
не думал про